bika/lims/exportimport/instruments/resultsimport.py

```diff
@@ -753,6 +753,44 @@ def _getZODBAnalysesFromReferenceAnalyses(self, objid, criteria):
 
         return analyses
 
+    def analysisHasCalcution(self, objid, analysis):
+        """ returns True or False"""
+
+        analyses = self._getZODBAnalyses(objid)
+        analyses_with_calculation = filter(lambda c: c.getCalculation(),analyses)
+        if analyses_with_calculation:
+            return True
+        return False
+
+    def getAnalysesWithCalculation(self, objid, analysis, analyses):
+        if self.analysisHasCalcution(objid, analysis):
+            analyses_with_calculation = filter(lambda c: c.getCalculation(),analyses)
+            return analyses_with_calculation
+
+    def calculateTotalResults(self, objid, analysis):
+        if self.analysisHasCalcution(objid, analysis):
+            # Analysis With Calculation
+            analyses = self._getZODBAnalyses(objid)
+            analyses_with_calculation = self.getAnalysesWithCalculation(objid, analysis, analyses)
+            for analysis_with_calc in analyses_with_calculation:
+                calcultion = analysis_with_calc.getCalculation()
+                formular = calcultion.getMinifiedFormula()
+                # The analysis that we are currenly on
+                analysis_keyword = analysis.getKeyword()
+                if analysis_keyword in formular:
+                    calc_passed = analysis_with_calc.calculateResult(override=self._override[1])
```

Comment:
Please add a one line comment above. This helps the reviewer to understand what this piece of code does w/o the need to inspect the whole file. You know the code better, help us to understand what you found out Lunga.